tests/blackbox/stratisd_cert.py

```diff
@@ -345,6 +345,50 @@ def test_pool_add_cache_permissions(self):
             StratisDbus.pool_add_cache, [pool_path, StratisCertify.DISKS[2:3]], True
         )
 
+    def test_pool_create_after_cache(self):
+        """
+        Test creating existing pool after cache was added
+        """
+        pool_name = p_n()
+        pool_path = make_test_pool(pool_name, StratisCertify.DISKS[0:1])
+
+        self._unittest_command(
+            StratisDbus.pool_init_cache(pool_path, StratisCertify.DISKS[1:2]),
+            dbus.UInt16(0),
+        )
+        self._unittest_command(
+            StratisDbus.pool_create(pool_name, StratisCertify.DISKS[0:1], None),
+            dbus.UInt16(0),
+        )
+
+    def test_pool_create_with_cache(self):
+        """
+        Test creating existing pool with device already used by cache fails
+        """
+        pool_name = p_n()
+        pool_path = make_test_pool(pool_name, StratisCertify.DISKS[0:1])
+
+        self._unittest_command(
+            StratisDbus.pool_init_cache(pool_path, StratisCertify.DISKS[1:2]),
+            dbus.UInt16(0),
+        )
+        self._unittest_command(
+            StratisDbus.pool_create(pool_name, StratisCertify.DISKS[0:2], None),
+            dbus.UInt16(1),
+        )
+
+    def test_pool_create_cache_with_data_device(self):
```

Comment:
This test is testing something different. I'm not sure if this is related.

Reply:
I think _this_ one is unrelated. What would happen here is that the Stratis metadata would be found on the data devices that were being proposed as cache devices, and so they would be rejected. This would occur in BlockDevMgr::initialize.